docs/source/aws/contribute.rst

```diff
@@ -0,0 +1,169 @@
+.. _aws_contribute:
+
+Developer Guide
+=================
+
+Cloud Custodian is a Python application and supports Python 2 and 3 on Linux and Windows. It is recommended 
+using Python 3.6 or higher.
+
+Run the following commands in the root directory after cloning Cloud Custodian:
+
+.. code-block:: bash
+
+    $ make install
+    $ source bin/activate
+
+This creates a virtual env in your enlistment and installs all packages as editable.
+
+Now you may run ``python3 c7n/cli.py`` with any flags in order to directly test changes to the source files.  For example, 
+``python3 c7n/cli.py schema aws.<resource_type>`` will return schema for resource type.
+
+
+Adding New AWS Resources
+==========================
+
+Create New AWS Resource
+-------------------------
+
+Each class definition will use the ``@resources.register('<resource_name>')`` decorator to register that class as a Custodian resource 
+substituting `<resource_name>` with the new resource name. The name specified in the decorator is how the resource will be referenced 
+within policies.
+
+Register the new resource: ``@resources.register(‘<resource_name>’)``
+
+An outer class defining the reference in resource mapping: ``class <resource_type>(query.QueryResourceManager)``
+
+Interior class that defines the aws metadata for resource
+``class resource_type(query.TypeInfo)``:
+
```

Comment:
so the actual documentation reference and which this content should be added to is the TypeInfo for query.py,

ideal is if we make it a class doc string there and then pull the doc here via sphinx.

Reply:
Hi Kapil, thanks for all of your comments! 
I have not been able to find any references to query.py anywhere else in the documentation. Could you provide more information on how to make query.py a class docstring so it can be referenced elsewhere using sphinx?

Alternatively, in our updated PR we included the documentation you mentioned from query.py as a code block in our contribute.rst. Let us know if it this ok or if you'd rather we follow your initial suggestion.

Thanks,
Liat

Reply:
i guess i was looking to add the docs that are in this new developer doc, back to query.py ie wrt to the structure of say enum_spec and detail_spec.. for those changes to be useful in this guide it would need a refactor of the docs in query.py to use a class doc string, and then include them back to this document via sphinx directive.

Reply:
rethinking this, i think we should probably separate out the dev docs into a separate pr, as the rest of this looks close to ready to go, but the docs are going to need a bit more cleanup and are unrelated to the rest of this feature pr.

Reply:
Hi Kapil - this PR has been updated (AWS dev docs removed).
Once the dev docs are ready, we'll submit a new PR.